docs: add advanced usage of colcon (#27)
autowarefoundation/autoware-documentation: docs/how-to-guides/advanced-usage-of-colcon.md

```

```bash
colcon build --cmake-args -DCMAKE_BUILD_TYPE=Release
```

## Changing the default configuration of colcon

Create `$COLCON_HOME/defaults.yaml` to change the default configuration.

```bash
mkdir -p ~/.colcon
cat << EOS > ~/.colcon/defaults.yaml
{
    "build": {
        "symlink-install": true
    }
}
```

See [here](https://colcon.readthedocs.io/en/released/user/configuration.html#defaults-yaml) for more details.

## Generating compile_commands.json

[compile_commands.json](https://colcon.readthedocs.io/en/released/user/how-to.html#cmake-packages-generating-compile-commands-json) is used by IDEs/tools to analyze the build dependencies and symbol relationships.

You can generate it with the flag `DCMAKE_EXPORT_COMPILE_COMMANDS=1`:

```bash
colcon build --cmake-args -DCMAKE_EXPORT_COMPILE_COMMANDS=1
```

## Seeing compiler commands

To see the compiler and linker invocations for a package, use `VERBOSE=1` and `--event-handlers console_cohesion+`:

```bash
VERBOSE=1 colcon build --packages-up-to <package_name> --event-handlers console_cohesion+
```

See [here](https://colcon.readthedocs.io/en/released/reference/event-handler-arguments.html) for other options.

> ## Using Ccache

Author: @xmfcx FYI: Added to the dictionary by this PR.
https://github.com/tier4/autoware-spell-check-dict/pull/224

Author: And the spell-check CI has passed if I run it again.
https://github.com/autowarefoundation/autoware-documentation/runs/5218266236?check_suite_focus=true

Author: By the way, if you want to see files without being disturbed by some comments, you can try either:

1. Click the `View file` button
![image](https://user-images.githubusercontent.com/31987104/154294797-ec1efcb1-a560-447c-91d6-c4cc7eb2d019.png)

2. Use the rich view
![image](https://user-images.githubusercontent.com/31987104/154294726-ae6e9b94-bc5c-4b3b-b481-47aa7d931323.png)

[Ccache](https://ccache.dev/) can speed up recompilation.

> It is recommended to use it to save your time if there are no particular reasons.

Reviewer: ```suggestion
It is recommended to use it to save your time unless you have a specific reason not to do so.
```

Author: Thank you!
